support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/util/OAuth20Utils.java

```diff
@@ -487,4 +489,32 @@ public static String getClientIdFromAuthenticatedProfile(final CommonProfile pro
         val requestedClaims = parseRequestClaims(context);
         return requestedClaims.getOrDefault("userinfo", new HashMap<>()).keySet();
     }
+
+    /**
+     * Encode access token depending on contents of JWT.
+     *
+     * @param accessToken Access token to encode into a response
+     * @return Either the JWT or the access token's ID
+     */
+    public static String encodeAccessToken(final AccessToken accessToken) {
+        if (accessToken.getJwt() != null) {
+            return accessToken.getJwt();
+        }
+        return accessToken.getId();
+    }
+
+    /**
+     * Gets the access token id from a request.
+     *
+     * @param accessToken String either AT-... or the jwt access token
+     * @param oAuthAccessTokenIdExtractor service to decode the access token if jwt
+     * @return Access token id
+     */
+    @SneakyThrows
+    public static String getAccessTokenId(final String accessToken, final OAuthAccessTokenIdExtractor oAuthAccessTokenIdExtractor) {
```

Comment:
Should be unnecessary. Simply remove the method, and call `oAuthAccessTokenIdExtractor.extractId(accessToken);` wherever you need. The extractor should do the actual extraction on its own.